### DISCUSSION TITLE
Custom Domain showing blank page

### DISCUSSION THREAD

**Discussion by kamillaallimak** [community/community#48138] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi, I've been trying to configure a custom domain (somostango.se) for my website (https://github.com/kamillaallimak/somos).

I've followed the instructions both from provided by both Github, and from Loopia (my DNS provider). According to Loopia the fault should be from Github's side, since they say everything looks good and that somostango.se is pointing to my github project.

When try to confirm that my DNS record's are configured correctly (same for www.somostango.se):

`dig somostango.se

; <<>> DiG 9.10.6 <<>> somostango.se
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6073
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;somostango.se. IN A

;; ANSWER SECTION:
somostango.se. 2840 IN A 185.199.110.153
somostango.se. 2840 IN A 185.199.111.153
somostango.se. 2840 IN A 185.199.108.153
somostango.se. 2840 IN A 185.199.109.153

;; Query time: 16 msec
;; SERVER: 79.142.240.53#53(79.142.240.53)
;; WHEN: Mon Feb 20 18:35:46 CET 2023
;; MSG SIZE rcvd: 106
`
And according to githubs's Virtual assistant "Everything looks good on somostango.se!". Which seems reasonable since kamillaallimak.github.se redirects to somostango.se.

So both Github config's and Loopi config's seems to be correct, but something must be wrong somewhere since the page somostango.se shows nothing, anyone that can help me pinpoint the problem?

Thanks in advance!

Best regards
Kamilla (n00b programmer)

**Comment by kenjitagawa** [comment]
Hey Kamilla,

What does your Pages config looks like? From what I see of your project structure, you have a `docs` directory and a `src` directory. From what I know, your GitHub Pages will look for your project (i.e. the index.html) in one of two places, either the root `/` of the project, or the `/docs` directory. In any case, that also depends on you to choose which one.

**Comment by kamillaallimak** [comment]
Hi @kenjitagawa , the GitHub Pages site is being built from the gh-pages branch looking for it in /root . It used to work before I added the custom domain.

**Comment by kenjitagawa** [comment, marked as answer]
That leads me to think that there may be some misconfiguration with your DNS. Based on my current configuration, I have the following structure:

|  Hostname  |  Type  |  TTL  |  DATA  |
|---|---|---|---|
|  `domain.com`  |  A  |  3600  |  185.199.108.153<br>185.199.109.153<br>185.199.110.153<br>185.199.111.153 |
|  github-pages-challenge  |  TXT  |  3600  |  key-provided-by-gh  |
| `www.domain.com` | CNAME | 3600 | `yourusername.github.io.` |

**Comment by kamillaallimak** [comment]
@kenjitagawa I see, this is my DNS config:
<img width="885" alt="image" src="https://user-images.githubusercontent.com/54072632/220348799-567191a4-0e6f-43f0-a2b3-1dd1783693da.png">

What I see missing is the guthub-pages-challenge part, is that necessary?

**Comment by kenjitagawa** [comment]
Have you added the custom domain to your GitHub account? When it comes to adding the domain to your project, GitHub asks for the verification of this domain, the way you can accomplish that, is by going to the settings of your personal account, and then add a domain. Once there, GitHub will give you a DNS TXT record to add which is going to be your secret. 

1. Go to:
[Pages settings](https://github.com/settings/pages) and `Add a domain`

2. Then, add the TXT record to your provider's page
[Add it here](https://github.com/settings/pages_verified_domains/somostango.se)

**Reply by kamillaallimak** [reply]
Oh, I had totally missed this part. 
I have added it, and it has now been verified from Github's side:

<img width="651" alt="image" src="https://user-images.githubusercontent.com/54072632/220379427-b66f3b35-2a16-4b2c-9340-1b7377b40b1d.png">

<img width="839" alt="image" src="https://user-images.githubusercontent.com/54072632/220376081-1c35795b-02cb-4f42-ade2-9a0e9a52283b.png">

However, the page is still blank, and "dig somostango.se" / "dig www.somostango.se" still shows the same stuff as earlier:

`www.somostango.se.	2791	IN	CNAME	kamillaallimak.github.io.

kamillaallimak.github.io. 2791	IN	A	185.199.109.153

kamillaallimak.github.io. 2791	IN	A	185.199.111.153

kamillaallimak.github.io. 2791	IN	A	185.199.110.153

kamillaallimak.github.io. 2791	IN	A	185.199.108.153`


Is something still missing?

(Also, thanks for the support @kenjitagawa , I owe you!)

**Reply by kenjitagawa** [reply]
No problem!

Using the tool [Pages-health-check](https://github.com/github/pages-health-check), I've compared yours to a working configuration of mine, and here are the differences:

Yours:
```yaml
nameservers: :default
dns_resolves?: true
proxied?: false
cloudflare_ip?: false
fastly_ip?: false
old_ip_address?: false
a_record?: true
aaaa_record?: false
a_record_present?: true
aaaa_record_present?: false
cname_record?: false
mx_records_present?: true
valid_domain?: true
apex_domain?: true
should_be_a_record?: true
cname_to_github_user_domain?: false
cname_to_pages_dot_github_dot_com?: false
cname_to_fastly?: false
pointed_to_github_pages_ip?: true
non_github_pages_ip_present?: false
pages_domain?: false
served_by_pages?: true
valid?: true
reason:
https?: true
enforces_https?: true
https_error:
https_eligible?: true
caa_error:
dns_zone_soa?: true
dns_zone_ns?: true
```

Mine:

```yaml
nameservers: :default
dns_resolves?: true
proxied?: false
cloudflare_ip?: false
fastly_ip?: false
old_ip_address?: false
a_record?: false
aaaa_record?: false
a_record_present?: false
aaaa_record_present?: false
cname_record?: true
mx_records_present?: false
valid_domain?: true
apex_domain?: false
should_be_a_record?: false
cname_to_github_user_domain?: true
cname_to_pages_dot_github_dot_com?: false
cname_to_fastly?: false
pointed_to_github_pages_ip?: false
non_github_pages_ip_present?: false
pages_domain?: false
served_by_pages?: true
valid?: true
reason:
https?: true
enforces_https?: false
https_error:
https_eligible?: true
caa_error:
dns_zone_soa?: false
dns_zone_ns?: false
```

We have several things that differ. 

1. Yours has an A record where mine does not.
2. Mine has a CNAME where yours does not. 
3. You have an MX, for email, not that I should have it, but could be a factor for possible issues, maybe? Maybe causing some conflicts?
4. You are using an Apex domain. Which is the `@` that you have set up. It serves a bit of a catch all to the current origin. It doesn't have a name from my knowledge since it's something that some DNS providers use whereas others use just an empty string. You can read a bit more about it on this thread on [Serverfault](https://serverfault.com/questions/426326/what-is-the-name-for-a-dns-record-starting-with#:~:text=The%20%40%20symbol%20in%20your%20DNS,www%20or%20sub%2Ddomain%20name.)
5. One thing that stands out quite a bit to me, is that your `cname_to_github_user_domain` is currently `false`. 

Keeping in mind that;
1. Mine is a simple homelab documentation
2. Mine is using a subdomain

For organization sake, and also possibly for some troubleshoooting, I would separate the following group into three groups.
![image](https://user-images.githubusercontent.com/61639117/220466613-d3eba00f-f9b2-4c72-b865-dc252c988714.png)

Here is what I suggest instead:

|  Hostname  |  Type  |  TTL  |  DATA  |
|---|---|---|---|
|  `domain.com`  |  A  |  3600  |  185.199.108.153<br>185.199.109.153<br>185.199.110.153<br>185.199.111.153 |
| `domain.com` | MX |  3600 | {{ DATA }} |
| `domain.com` | NS |  3600 | {{ DATA }} |

Additionally, I would check about the `loopia.com` stuff:
* The MX is a mailing server, which I presume that you use and in either case, shouldn't impact your website.
* The SRV is a VoIP, which I would check for the chances of that causing issues on Loopia's website and documentation. 
* Another question that I would ask is about the other CNAME that you have there also belonging to Loopia, could that be causing your problem?
![image](https://user-images.githubusercontent.com/61639117/220467329-5670cc9e-06c3-42d8-b0d3-38b26fd8f1f4.png)

If ever these don't work, let me know. Additionally, you can add me on Discord if you would like and we can try to go over the settings on GitHub and the DNS config. (Kenjitagawa#2406)

**Reply by yrncollo** [reply]
> Oh, I had totally missed this part. I have added it, and it has now been verified from Github's side:
> 
> <img alt="image" width="651" src="https://user-images.githubusercontent.com/54072632/220379427-b66f3b35-2a16-4b2c-9340-1b7377b40b1d.png">
> 
> <img alt="image" width="839" src="https://user-images.githubusercontent.com/54072632/220376081-1c35795b-02cb-4f42-ade2-9a0e9a52283b.png">
> 
> However, the page is still blank, and "dig somostango.se" / "dig [www.somostango.se](http://www.somostango.se)" still shows the same stuff as earlier:
> 
> `www.somostango.se. 2791 IN CNAME kamillaallimak.github.io.
> 
> kamillaallimak.github.io. 2791 IN A 185.199.109.153
> 
> kamillaallimak.github.io. 2791 IN A 185.199.111.153
> 
> kamillaallimak.github.io. 2791 IN A 185.199.110.153
> 
> kamillaallimak.github.io. 2791 IN A 185.199.108.153`
> 
> Is something still missing?
> 
> (Also, thanks for the support @kenjitagawa , I owe you!)

So how long did it take for it to be verified? I think I am on the same situation you were before you resolved this

**Reply by kenjitagawa** [reply]
> > Oh, I had totally missed this part. I have added it, and it has now been verified from Github's side:
> 
> > 
> 
> > <img alt="image" width="651" src="https://user-images.githubusercontent.com/54072632/220379427-b66f3b35-2a16-4b2c-9340-1b7377b40b1d.png">
> 
> > 
> 
> > <img alt="image" width="839" src="https://user-images.githubusercontent.com/54072632/220376081-1c35795b-02cb-4f42-ade2-9a0e9a52283b.png">
> 
> > 
> 
> > However, the page is still blank, and "dig somostango.se" / "dig [www.somostango.se](http://www.somostango.se)" still shows the same stuff as earlier:
> 
> > 
> 
> > `www.somostango.se. 2791 IN CNAME kamillaallimak.github.io.
> 
> > 
> 
> > kamillaallimak.github.io. 2791 IN A 185.199.109.153
> 
> > 
> 
> > kamillaallimak.github.io. 2791 IN A 185.199.111.153
> 
> > 
> 
> > kamillaallimak.github.io. 2791 IN A 185.199.110.153
> 
> > 
> 
> > kamillaallimak.github.io. 2791 IN A 185.199.108.153`
> 
> > 
> 
> > Is something still missing?
> 
> > 
> 
> > (Also, thanks for the support @kenjitagawa , I owe you!)
> 
> 
> 
> So how long did it take for it to be verified? I think I am on the same situation you were before you resolved this

The issue was not with the DNS, the configuration marked as the answer is the one that you should have with your DNS provider. The issue, we came to find was during the build process of Angular and where each file was placed. Want to share more details about your issue? Maybe we can help! @yrncollo

**Reply by MadeehaKhan** [reply]
@kenjitagawa I am currently facing the same problem even though my configurations look fine. I'd really appreciate some help :)

**Reply by kenjitagawa** [reply]
@MadeehaKhan Sorry I took a while! Are you still facing the issue? If so, do you want to create a new discussion and add some more information about your current setup and configuration, and tag me on it?

**Reply by MadeehaKhan** [reply]
@kenjitagawa I was able to figure it out, but thank you so much for getting back to me!
